server/implementation-cdi/src/main/java/io/smallrye/graphql/cdi/context/SmallRyeContextAccessorProxy.java

```diff
@@ -0,0 +1,302 @@
+package io.smallrye.graphql.cdi.context;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Specializes;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+
+import graphql.ExecutionInput;
+import graphql.schema.DataFetchingEnvironment;
+import io.smallrye.graphql.execution.QueryCache;
+import io.smallrye.graphql.execution.context.DocumentSupplier;
+import io.smallrye.graphql.execution.context.SmallRyeContext;
+import io.smallrye.graphql.execution.context.SmallRyeContextManager;
+import io.smallrye.graphql.schema.model.Field;
+
+/**
+ * Class that serves as a proxy decorator for obtaining the current SmallRyeContext. It always calls
+ * SmallRyeContext.getContext() to obtain the current instance, and delegates to it.
+ * This is necessary because the SmallRyeContext is an immutable class, yet we, in some cases,
```

Comment:
I'm not sure where this note is coming from (I think it was there before) but it's clearly NOT immutable. The part that we need to inject different instances during one request, and more importantly, even into the same `@GraphQLApi` endpoint instance, is still true. 

Could you rephrase it so that it is clear that it's needed because we need a single `@GraphQLApi` instance be able to see different instances of the context? (source method needs to see a different context than the operation root)

Reply:
OK I removed all meta data. So users can still unwrap and get to dataFetcherEnv and add there, but not directly anymore.